ts/util/saveURLAsFile.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * @prettier
+ */
+export const saveURLAsFile = ({
+  filename,
+  url,
+  document,
+}: {
+  filename: string;
+  url: string;
+  document: Document;
+}): void => {
+  const anchorElement = document.createElement('a');
+  anchorElement.href = url;
+  anchorElement.download = filename;
+  anchorElement.click();
```

Comment:
Interesting technique here. I feel like it may need some comments.

Reply:
I wish I could take credit for it but it’s a port of the `saveFile` function buried inside `AttachmentView`: https://github.com/signalapp/Signal-Desktop/blob/45bf1d5bfcaf08f39695d11e4e1a6473147f0a90/js/views/attachment_view.js#L200-L205

Let me know what comment would be helpful.